### PR TITLE
onHandlesReady waits for planHandles

### PR DESCRIPTION
**Pull request by cromwellian**
Discovered during hackathon

### CODE REVIEW

**Review by jblebrun — APPROVED** [review]

**Comment by jblebrun** [comment]
Great find!

File: java/arcs/core/host/ArcHostContextParticle.kt [hunk 1]
```diff
@@ -174,11 +174,13 @@ class ArcHostContextParticle(
     val onReadyJobs = mapOf(
       "particles" to Job(),
       "arcHostContext" to Job(),
-      "handleConnections" to Job()
+      "handleConnections" to Job(),
```
Comment (shaper) [inline java/arcs/core/host/ArcHostContextParticle.kt:177]: Is there a test we can add that would have caught this?

Comment (mykmartin) [reply java/arcs/core/host/ArcHostContextParticle.kt:177]: Probably not. The more long-term solution would be to finish up the work I was trying to get through before moving back to the TS code: make non-particle handles automatically synchronised so these types of wait checks (including all the `awaitReady` hooks) are no longer needed.

**Review by mykmartin — APPROVED** [review]

**Pull request closed without merging**